misc.go

```diff
@@ -42,16 +42,16 @@ func (t SlackResponse) Err() error {
 
 // StatusCodeError represents an http response error.
 // type httpStatusCode interface { HTTPStatusCode() int } to handle it.
-type statusCodeError struct {
```

Comment:
this is not necessary.

library consumers can use:
```
type statuscode interface {
   HTTPStatusCode() int
}
if x, ok := err.(statuscode); ok {
}
```

Reply:
This seems pretty cumbersome and not idiomatic to me

Reply:
I tend to agree w/ [dave cheney on errors](https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully).

and so does the golang team if you look at their proposed changes to error handling coming up.

Reply:
as an aside, I'm happy to alleviate some of the usability issues of the statuscode error in other ways. such as having the interface type exposed in slackutils package (with proper tests tying it back to statuscode so when I invariably break it a test warns us)

Reply:
Thanks for the link, TIL I've been doing error types incorrectly. The specific usability issue we're having is exactly the one in Dave Cheney's example: deciding when to retry an error from Slack. I think a Retryable interface would solve that. Will give that a go.

The slackutils package does give me some pause... https://blog.golang.org/package-names

Reply:
the slackutils package is suppose to give you pause. but there is precedence see httputils etc.

slackutils is for stuff that isn't well thought out, or well structured enough to put into the core library.